core/src/main/scala/org/apache/spark/deploy/raydp/RayAppMaster.scala

```diff
@@ -262,6 +277,14 @@ class RayAppMaster(host: String,
       val classPathEntries = appInfo.desc.command.classPathEntries.mkString(";")
       RayExecutorUtils.setUpExecutor(handlerOpt.get, appId, driverUrl, cores, classPathEntries)
     }
+
+    private def getNextBundleIndex: Int = {
+      if (placementGroup != null && bundleIndexes.nonEmpty) {
+        val previous = currentBundleIndex
+        currentBundleIndex = (currentBundleIndex + 1) % bundleIndexes.size
+        previous
+      } else -1
+    }
```

Comment:
What if this is called more than expected? Is this compatible with dynamic executor allocation?

Reply:
I guess no. Also we should update the doc and comment

Reply:
No, this should only be a temporay solution. The placement group's python API supports using any available bundle without specifying the bundle index, but there is no such API in Java. I'm trying to upstream the Java API, and we can remove this part of code to let Ray decide which bundle to use when the API become available.